Userland/cat.cpp

```diff
@@ -25,27 +25,55 @@
  */
 
 #include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/stat.h>
 #include <string.h>
 #include <unistd.h>
 
+bool detect_useless_use_of_cat(int argc)
+{
+    // I can think of somewhat valid uses of zero-argument cat
+    // from a tty to a pipe, so let's allow those.
+    if (argc != 2)
+        return false;
+
+    struct stat statbuf;
+    int rc = fstat(1, &statbuf);
+    if (rc < 0)
+        return false;
+
+    return S_ISFIFO(statbuf.st_mode);
+}
+
 int main(int argc, char** argv)
 {
     if (pledge("stdio rpath", nullptr) < 0) {
         perror("pledge");
         return 1;
     }
 
+    Vector<const char*> paths;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(argc, argv);
+
+    if (detect_useless_use_of_cat(argc))
+        fprintf(stderr, "\033[34;1mMeow! Useless use of cat detected\033[0m\n");
```

Comment:
This seems like an arbitrary thing to condescend to the user about?

Reply:
but...isn't it an adorable thing to do?
and it's harmless anyway?

Reply:
It's not arbitrary, it's such [a part of the Unix culture](http://porkmail.org/era/unix/award.html)...

Reply:
Perhaps it's worth tweaking the message to make it sound as a friendly suggestion rather than condescending?

First attempt:

"Meow! Hey, you can simply use file redirection (< `argv[1]`) instead of spawning a cat process"

Reply:
Oh, I was not aware of this old Unix meme :)
With that in mind, I think it makes perfect sense to word it like you did.